### PR TITLE
orca: cleanup old code, and get grpc package to use new code

### PR DESCRIPTION
**Pull request by easwars**
Fixes https://github.com/grpc/grpc-go/issues/5542

This PR cleans up the old implementation in https://github.com/grpc/grpc-go/tree/master/xds/internal/balancer/orca and gets the `grpc` package to use the new one (by registering the `Parser` interface from the new code).

RELEASE NOTES: n/a

### CODE REVIEW

**Review by dfawley — COMMENTED** [review]
Was there an end-to-end style test for this that creates a gRPC client with a custom LB policy that retrieves ORCA data from the server?

File: orca/orca.go [hunk 1]
```diff
@@ -162,3 +163,21 @@ func ToLoadReport(md metadata.MD) (*v3orcapb.OrcaLoadReport, error) {
...
+// loadParser implements the Parser interface defined in `internal/balancerload`
+// package. This interface is used by the client stream to parse load reports
+// sent by the server in trailer metadata. The parsed loads are then sent to
+// balancers via balancer.DoneInfo.
+//
+// The grpc package cannot directly call orca.ToLoadReport() as that would cause
+// an import cycle. Hence this roundabout method is used.
+type loadParser struct{}
+
+func (*loadParser) Parse(md metadata.MD) interface{} {
```
[the remark this answers is not in the document]
Comment (easwars) [reply orca/orca.go]: go/go-style/decisions#receiver-type
I have read this a few times in the past, but only few things stuck in my head like:
- If the method needs to mutate the receiver, the receiver must be a pointer.
- If the receiver is a struct containing fields that cannot safely be copied, use a pointer receiver.
- When in doubt, use a pointer receiver.

But there seems to be case for this one:
- If the receiver is a "small" array or struct that is naturally a value type with no mutable fields and no pointers, a value receiver is usually the right choice.

File: orca/orca.go [hunk 2]
```diff
@@ -162,3 +163,21 @@ func ToLoadReport(md metadata.MD) (*v3orcapb.OrcaLoadReport, error) {
...
+	lr, _ := ToLoadReport(md)
```
[the remark this answers is not in the document]
Comment (easwars) [reply orca/orca.go]: Added an error log. As far as the caller of this function is concerned (transport), the error does not change anything. So, I did not bother changing the signature of the method to return an error for the caller to deal with it.

**Comment by easwars** [comment]
> Was there an end-to-end style test for this that creates a gRPC client with a custom LB policy that retrieves ORCA data from the server?

The [e2e style test](https://github.com/grpc/grpc-go/blob/master/orca/call_metric_recorder_test.go) that we have for this does create a gRPC client, but simply reads the trailers off the stream instead of installing a custom LB policy. I thought it would be better/easier to add that test once we have client-side support as well.

**Review by dfawley — APPROVED** [review]
One small cleanup item, otherwise LGTM

File: xds/internal/balancer/clusterimpl/picker.go [hunk 3]
```diff
@@ -158,7 +159,7 @@ func (d *picker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
 			}
 			d.loadStore.CallFinished(lIDStr, info.Err)
 
-			load, ok := info.ServerLoad.(*orcapb.OrcaLoadReport)
+			load, ok := info.ServerLoad.(*v3orcapb.OrcaLoadReport)
```
[the remark this answers is not in the document]
Comment (easwars) [reply xds/internal/balancer/clusterimpl/picker.go:162]: Done. Thanks.

**Pull request merged**